apps/common/components/Table.tsx

```diff
@@ -0,0 +1,97 @@
+
+import type {ReactElement} from 'react';
+
+type TSortOrder = 'asc' | 'desc' | '';
+
+type TState<T> = {
+	sortedBy: Extract<keyof T, string> | undefined, 
+	order: TSortOrder
+}
+
+const switchOrder = (order: TSortOrder): TSortOrder => order === '' ? 'desc' : order === 'desc' ? 'asc' : '';
+
+type TMetadata<T> = {
+	key: Extract<keyof T, string>;
+	label: string;
+	className?: string;
+	sortable?: boolean;
+	fullWidth?: boolean;
+	columnSpan?: number;
+	format?: (item: T) => string;
+	transform?: (item: T) => ReactElement;
+}
+  
+type TTableProps<T> = {
+	metadata: TMetadata<T>[];
+	data: T[];
+	columns?: number;
+	initialSortBy?: Extract<keyof T, string>;
+	onRowClick?: (item: T) => void;
+}
+
+function Table<T>({metadata, data, columns, initialSortBy, onRowClick}: TTableProps<T>): ReactElement {
+	const [{sortedBy, order}, set_state] = useState<TState<T>>({sortedBy: initialSortBy, order: 'desc'});
+	
+	const handleSort = useCallback((key: Extract<keyof T, string>): void => {
+		const willChangeSortKey = sortedBy !== key;
+		const newOrder = switchOrder(willChangeSortKey ? '' : order);
+		set_state({sortedBy: newOrder ? key : undefined, order: newOrder});
+	}, [order, sortedBy]);
+    
+	const sortedData = useMemo((): T[] => {
+		return sortedBy && order ? sort(data, sortedBy, order) : data;
+	}, [data, order, sortedBy]);
+
+	const numberOfColumns = Math.min(columns ?? (metadata.length), 12);
+
+	return (
+		<div className={'w-full'}>
+			<div className={`mb-2 hidden w-full px-8 md:grid ${`md:grid-cols-${numberOfColumns}`}`}>
+				{metadata.map(({key, label, sortable, className, columnSpan}): ReactElement => (
+					<button
+						key={`header_${key}`} 
+						onClick={(): void => sortable ? handleSort(key) : undefined}
+						disabled={!sortable}
+						className={`flex flex-row items-center justify-end space-x-1 first:justify-start ${`md:col-span-${columnSpan ?? 1}`} ${className || ''}`}
+					>
+						<p className={'text-base text-neutral-400'}>
+							{label}
+						</p>
+						{sortable && sortedBy === key && <IconChevronPlain className={`yearn--sort-chevron ${order === 'asc' ? 'rotate-180' : ''}`} />}
+						{sortable && sortedBy !== key && <IconChevronPlain className={'yearn--sort-chevron--off text-neutral-300 group-hover:text-neutral-500'} />}
+					</button>
+				))}
+			</div>
+			
+			{sortedData.map((item, rowIndex): ReactElement => {
+				return (
+					<div 
+						key={`row_${rowIndex}`} 
+						className={`grid w-full grid-cols-1 border-t border-neutral-200 px-4 py-2 transition-colors hover:bg-neutral-300 md:border-none md:px-8 ${`md:grid-cols-${numberOfColumns}`} ${onRowClick ? 'cursor-pointer' : 'cursor-auto'}`}
+						onClick={(): void => onRowClick?.(item)}
+					>
+						{metadata.map(({key, label, className, fullWidth, columnSpan, format, transform}): ReactElement => {
+							// eslint-disable-next-line @typescript-eslint/no-explicit-any
+							const isNumber = !isNaN(item[key] as any);
```

Comment:
I reckon here we could do just cast `item[key]` to number, as that's what `isNaN` wants, and if it's not then it would just return `true`. So we should be safe

```suggestion
							const isNumber = !isNaN(item[key] as number);
```

Reply:
Interesting, I don't get that error. Which version of TypeScript are you using?